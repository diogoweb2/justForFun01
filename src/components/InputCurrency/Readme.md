```js
<InputCurrency value="10" />
```
