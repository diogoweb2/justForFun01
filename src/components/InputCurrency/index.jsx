import React from "react";
import PropTypes from "prop-types";

const InputCurrency = ({ value, onChange }) => (
  <input onChange={e => onChange(e.target.value)} type="text" value={value} />
);

InputCurrency.propTypes = {
  value: PropTypes.number,
  onChange: PropTypes.func
};

InputCurrency.defaultProps = {
  value: 0,
  onChange: null
};

export default InputCurrency;
