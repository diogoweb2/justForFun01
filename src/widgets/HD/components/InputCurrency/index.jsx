import React from "react";
import PropTypes from "prop-types";

const InputCurrency = ({ value, onChange }) => (
  <input onChange={onChange} type="text" value={value} />
);

InputCurrency.propTypes = {
  value: PropTypes.number,
  onChange: PropTypes.func
};

InputCurrency.defaultProps = {
  value: 0,
  onChange: null
};

export default InputCurrency;
