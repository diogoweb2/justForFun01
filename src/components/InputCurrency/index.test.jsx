import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import InputCurrency from "./index";

let wrapper;
describe("InputCurrency", () => {
  it("should render", () => {
    wrapper = shallow(<InputCurrency value="test" />);
    expect(wrapper).toHaveLength(1);
  });

  it("should match snapshot...", () => {
    wrapper = shallow(<InputCurrency value="test" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
