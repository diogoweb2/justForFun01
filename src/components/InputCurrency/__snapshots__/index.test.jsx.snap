// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCurrency should match snapshot... 1`] = `
<input
  className="slds-input"
  disable="false"
  onChange={[Function]}
  placeholder="0.00"
  type="number"
  value="test"
/>
`;
